src/Servers/HttpSys/src/RequestProcessing/Request.cs

```diff
@@ -254,7 +254,7 @@ private AspNetCore.HttpSys.Internal.SocketAddress LocalEndPoint
         public string Scheme => IsHttps ? Constants.HttpsScheme : Constants.HttpScheme;
 
         // HTTP.Sys allows you to upgrade anything to opaque unless content-length > 0 or chunked are specified.
-        internal bool IsUpgradable => !HasEntityBody && ComNetOS.IsWin8orLater;
+        internal bool IsUpgradable => ProtocolVersion < HttpVersion.Version20 && !HasEntityBody && ComNetOS.IsWin8orLater;
```

Comment:
I know in-proc still always returns true for IsUpgradableRequest, but does HttpSys also not check for a `Connection: Upgrade` request header? If not, why not?

Reply:
Http.Sys doesn't require it, it allows you to upgrade anything that doesn't have a request body. Components like WebSockets do their own check for the headers they require.

Checking the spec (https://tools.ietf.org/html/rfc2616#section-14.42) it seems the only place the Upgrade header is required is on the 101 response. On the request it's a hint.

I wouldn't ask to make it optional in Kestrel unless someone had a good reason, but I don't see the need to add the check here if it works without it.

Reply:
Am I correct to assume Http.Sys does require an HTTP/1.x connection to upgrade?

Reply:
Yes, otherwise it returns ERROR_INVALID_PARAMETER.